### DISCUSSION TITLE
New to docker, not sure if this is a docker issue, or a photoprism issue?  Nothing in storage or database folder.

### DISCUSSION THREAD

**Discussion by surfrock66** [photoprism/photoprism#2736] [General]
I don't typically use docker, but I'm deploying photoprism using the docker-compose.yml provided.  I created /etc/photoprism which is owned by root with permissions 755. I've only updated 6 lines:

- I've added "photoprism" as the container_name for photoprism
- I've added "photoprism_db" as the container_name for mariadb
- I've changed the admin password
- I added an originals directory
- I mapped "/etc/photoprism/storage" to "/photoprism/storage" in the photoprism container
- I mapped "/etc/photprism/database" to "/var/lib/mysql" in the mariadb container

When I start the container with "docker-container up -d" it starts, creates the "storage" and "database" directories, scans my originals folder, everything seems great.  The only issue is, when I change settings, it looks like they don't actually stay changed?  That made me investigate, and I am finding that my storage and database directories are completely empty.  If I stop and start the container with "docker-compose stop" then "docker-compose up -d" everything is still there, so it's not like the storage is wiped.  Where are things going here?  I figured the directories under /etc/photoprism would start getting huge.

**Comment by lastzero** [comment]
First, you should not store user data in /etc. Second, make sure you mount the folders correctly and specify the exact configuration you used for them. If you forgot something, it is possible that these folders were not mounted correctly. It's also possible that the app doesn't have permission to write to /etc. It may be helpful to check the Docker logs for this.

**Reply by lastzero** [reply]
👉 https://docs.photoprism.app/getting-started/troubleshooting/docker/

**Comment by surfrock66** [comment]
Gotcha.  I haven't done much with docker in the past, and I've read things and looked through general docker guides, but I don't have a good grasp of best practices (like don't put container data in /etc); I'll strive move the storage to opt.  

I found this in the syslogs:

```
Sep 28 08:07:47 hostname systemd[1]: run-docker-netns-94b99fa76efe.mount: Deactivated successfully.
Sep 28 08:07:47 hostname systemd[1]: var-lib-docker-overlay2-6f54dff5983027e02edeb7572db709c552acb5990fa4475df96898c5506fff3f-merged.mount: Deactivated successfully.
Sep 28 08:20:17 hostname dockerd[2992]: time="2022-09-28T08:20:17.097489176-07:00" level=warning msg="Security options with `:` as a separator are deprecated and will be completely unsupported in 17.04, use `=` instead."
Sep 28 08:20:17 hostname dockerd[2992]: time="2022-09-28T08:20:17.097530281-07:00" level=warning msg="Security options with `:` as a separator are deprecated and will be completely unsupported in 17.04, use `=` instead."
Sep 28 08:20:17 hostname systemd[1]: var-lib-docker-overlay2-c56bfbfea1903dea48c0589dda410ab252314e288752b832fb31eeaee0bd3e6b\x2dinit-merged.mount: Deactivated successfully.
```
I changed the : to = and suddenly I have the files.  Weird, but works.
